craft_providers/multipass/_ready.py

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+
+"""Multipass Provider Readiness Checks."""
+
+import logging
+
+from . import errors
+from .installer import is_installed
+from .multipass import Multipass
+
+logger = logging.getLogger(__name__)
+
+
+def ensure_multipass_is_ready(*, multipass: Multipass = Multipass()) -> None:
```

Comment:
why is this not an instance method in Multipass?

Reply:
This was to match the approach used with LXD, it certainly could (should?) be after considering it further.

Reply:
This change would result in cross coupling between installer module & Multipass